### PR TITLE
Hw4 task4 - fizzbuzz numbers and detailed instruction how to run doctests

### PR DESCRIPTION
**Pull request by IuliiaLanchuk**

### CODE REVIEW

File: homework4/task4/project4_doctests/task4_doctests.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        return [
+            "fizz buzz" * (i % 15 == 0)
+            or "fizz" * (i % 3 == 0)
+            or "buzz" * (i % 5 == 0)
+            or str(i)
+            for i in range(1, n + 1)
+        ]
+
+
+doctest.testmod()
```
Comment (Cjkjvfnby) [inline homework4/task4/project4_doctests/task4_doctests.py]: What is a recommended way to run it according to the documentation? And why?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: homework4/task4/project4_doctests/task4_doctests.py (hunk 1 above)
Comment (IuliiaLanchuk) [reply homework4/task4/project4_doctests/task4_doctests.py]: `if __name__ == "__main__":
    import doctest
    doctest.testmod()`
why...
__main__ point to scope, where code is executed. 
`if __name__ == "__main__":` call programm to execute, it is a runner, initiator.

**Comment by Cjkjvfnby** [comment]
I want to run all tests for all homework with a single command, how can I do it?

**Comment by IuliiaLanchuk** [comment]
@Cjkjvfnby `I want to run all tests for all homework with a single command, how can I do it?`
single command - `pytest`

**Comment by Cjkjvfnby** [comment]
> single command - pytest

1) why do you need  `if __name__ == "__main__":` then?
2) Pytest does not run doctests by default. Why does it work?

**Comment by IuliiaLanchuk** [comment]
As I understand both  `if __name__ == "__main__":  import doctest doctest.testmod()` and `pytest` command are test runners. So we can run tests by one of these ways. In pytest.ini fail should be line `pytest --doctest-modules`  to run doctest. I have this in my project. That is why it works. Also I have in pytest.ini fail `--doctest-continue-on-failure` that continue tests run even when we have failures.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**